packages/agents/src/agents/Chameleon/index.ts

```diff
@@ -86,20 +100,87 @@ export class ChameleonAgent extends Agent {
         },
         this.config.prompts.loopPreventionPrompt,
         this.config.prompts.agentSpeakPrompt,
-        this.beforeLlmResponse.bind(this)
+        this.beforeLlmResponse2.bind(this)
       );
     } catch (err) {
       this.context.logger.error(err);
       return ResultErr("Unrecoverable error encountered.");
     }
   }
 
-  protected initializeChat(args: GoalRunArgs): void {
+  protected initializeChat(): void {
     const { chat } = this.context;
-    
+
     chat.persistent(buildDirectoryPreviewMsg(this.context.workspace));
     chat.persistent("user", prompts.exhaustAllApproaches);
-    chat.persistent("user", args.goal);
```

Comment:
Added this back, removed it because the goal was getting duplicated (one here, one in initial messages). It should live here, and as you said we're moving away from initial messages.